hw/ip/otbn/util/shared/control_flow.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List, Set, Tuple
+
+from .decode import OTBNProgram, decode_elf
+from .insn_yaml import Insn
+from .section import CodeSection
+
+
+class ControlLoc:
+    '''Base class for control-flow locations.
+
+    Represents a normal PC that a branch or jump instruction could link to.
+    '''
+    def __init__(self, pc: int) -> None:
+        self.pc = pc
+
+    @classmethod
+    def is_special(cls) -> bool:
+        return False
+
+    def pretty(self) -> str:
+        return '{:#x}'.format(self.pc)
+
+
+class ImemEnd(ControlLoc):
+    '''Represents the end of instruction memory.'''
+    def __init__(self) -> None:
+        self.pc = -1  # dummy value
```

Comment:
Nit: Probably better to call the superclass explicitly? Something like:
```python3
    super().__init__(-1)
```
Then everything will work properly if you add something more to `ControlLoc`'s constructor in the future.

Reply:
Good point, fixed!